econml/inference.py

```diff
@@ -24,12 +43,74 @@ def __init__(self, n_bootstrap_samples=100, n_jobs=-1):
         self._n_bootstrap_samples = n_bootstrap_samples
         self._n_jobs = n_jobs
 
-    @property
-    def n_bootstrap_samples(self):
-        """Get how many draws to perform."""
-        return self._n_bootstrap_samples
+    def fit(self, estimator, *args, **kwargs):
+        est = BootstrapEstimator(estimator, self._n_bootstrap_samples, self._n_jobs, compute_means=False)
+        est.fit(*args, **kwargs)
+        self._est = est
+
+    def __getattr__(self, name):
+        if name.startswith('__'):
+            raise AttributeError()
+
+        m = getattr(self._est, name)
+
+        def wrapped(*args, alpha=0.1, **kwargs):
+            return m(*args, lower=100 * alpha / 2, upper=100 * (1 - alpha / 2), **kwargs)
+        return wrapped
+
+
+class StatsModelsInference(Inference):
+    """
+    Stores statsmodels covariance options.
+
+    This class can be used for inference by the LinearDMLCateEstimator.
+
+    Any estimator that supports this method of inference must implement a `statsmodels`
+    property that returns a `StatsModelsLinearRegression` instance and an `effect_op` property that returns an
+    `_EffectOp`.
+
+    Parameters
+    ----------
+    cov_type : string, optional (default 'HC1')
+        The type of covariance estimation method to use.  Supported values are 'nonrobust',
+        'fixed scale', 'HC0', 'HC1', 'HC2', and 'HC3'.  See the statsmodels documentation for
+        further information.
+
+    cov_kwds : optional additional keywords supported by the chosen method
+        Of the supported types, only 'fixed scale' has any keywords:
+        the optional keyword 'scale' which defaults to 1 if not specified.  See the statsmodels
+        documentation for further information.
+    """
+
+    def __init__(self, cov_type='HC1', use_t=False, **cov_kwds):
+        if cov_kwds and cov_type != 'fixed scale':
+            raise ValueError("Keyword arguments are only supported by the 'fixed scale' cov_type")
+        if cov_type not in ['nonrobust', 'fixed scale', 'HC0', 'HC1', 'HC2', 'HC3']:
+            raise ValueError("Unsupported cov_type; "
+                             "must be one of 'nonrobust', "
+                             "'fixed scale', 'HC0', 'HC1', 'HC2', or 'HC3'")
+
+        self.cov_type = cov_type
+        self.use_t = use_t
+        self.cov_kwds = cov_kwds
+
+    def prefit(self, estimator, *args, **kwargs):
+        self.statsmodels = estimator.statsmodels
+        # need to set the fit args before the estimator is fit
+        self.statsmodels.fit_args = {'cov_type': self.cov_type, 'use_t': self.use_t, 'cov_kwds': self.cov_kwds}
+
+    def fit(self, estimator, *args, **kwargs):
+        # once the estimator has been fit, it's kosher to access its effect_op and store it here
+        # (which needs to have seen the expanded d_t if there's a discrete treatment, etc.)
+        self.effect_op = estimator.effect_op
+
+    def effect_interval(self, X, *, T0=0, T1=1, alpha=0.1):
```

Comment:
I think that currently this works by coincidence. This function does not seem to be implementing the logic as a standalone. Basically: the treatments T0 and T1 are never expanded to vectors. The reason why it works is that because we are only using it in the dml cate estimator, which has overrident the effect and effect_interval of the super so as to first expand the treatment and then call the super effect and effect_interval. Then the super effect_interval sends to the statsmodelsinference effect_interval. But at that point treatments are already expanded to vectors. However, if we were to use it in some other cate estimator that does not expand the treatment and defers directly to inference, then it will be broken.

Reply:
The inference class should not need to expand the treatments because that should always be done by the estimator prior to calling this method.  I have therefore changed the signature of this method in the inference class to remove the defaults for `T0` and `T1` so that it does not imply that they can be scalars at this point.